browser/ipfs/ipfs_tab_helper.h

```diff
@@ -107,15 +108,32 @@ class IPFSTabHelper : public content::WebContentsObserver,
                            GatewayIPNS_No_Redirect_WhenNoDnsLink);
   friend class content::WebContentsUserData<IPFSTabHelper>;
   friend class BraveIPFSInfoBarDelegateObserverImpl;
+#if !BUILDFLAG(IS_ANDROID)  
+  friend class BraveIPFSFallbackInfoBarDelegateObserverImpl;
+
+  void SetSetShowFallbackInfobarCallbackForTesting(
+      base::RepeatingCallback<void(const GURL&)> callback) {
+    show_fallback_infobar_callback_for_testing_ = callback;
+  }
+#endif // !BUILDFLAG(IS_ANDROID)  
   explicit IPFSTabHelper(content::WebContents* web_contents);
 
   GURL GetCurrentPageURL() const;
   bool CanResolveURL(const GURL& url) const;
   bool IsDNSLinkCheckEnabled() const;
   bool IsAutoRedirectIPFSResourcesEnabled() const;
   void IPFSResourceLinkResolved(const GURL& ipfs);
-  void DNSLinkResolved(const GURL& ipfs, bool is_gateway_url);
-  void MaybeCheckDNSLinkRecord(const net::HttpResponseHeaders* headers);
+  void DNSLinkResolved(const GURL& ipfs,
+                       bool is_gateway_url
+#if !BUILDFLAG(IS_ANDROID)
```

Comment:
I think we can keep this argument for Android since tests will be failing

Reply:
done